src/db/models/leaderboard.ts

```diff
@@ -0,0 +1,50 @@
+import mongoose, { Document } from 'mongoose';
+
+export interface ITeamData {
+  team: string;
+  score: number;
+  attemptsLeft: number;
+  delta: string;
+}
+
+export interface ILeaderboard extends Document {
+  timestamp: Date;
+  data: ITeamData[];
+}
+
+const leaderboardSchema = new mongoose.Schema<ILeaderboard>(
+  {
+    timestamp: {
+      type: Date,
+      required: true,
+      unique: true,
+    },
+    data: {
+      type: [
+        {
+          team: { type: String, required: true },
+          score: { type: Number, required: true },
+          attemptsLeft: { type: Number, required: true },
+          delta: { type: String, required: true },
+        },
+      ],
+      required: true,
+    },
+  },
+  {
+    timestamps: true,
+    toJSON: {
+      transform: (doc, ret) => {
+        delete ret._id;
+        delete ret.__v;
+        return ret;
+      },
+    },
+  },
+);
+
+const Leaderboard =
+  mongoose.models.Leaderboard ||
+  mongoose.model<ILeaderboard>('Leaderboard', leaderboardSchema, 'leaderboard');
```

Comment:
Why are there 3 different names for the model ?

Reply:
In the DB, it is named `leaderboard`, but how mongoose reads the collection name is that it is converted to lowercase and plural, so with `leaderboards` it would have been 2 names in the model. But since in the db leaderboard is not plural, I need to have the third name as that is the one that bypasses the added plural.